src/omnisharp/server.ts

```diff
@@ -769,7 +775,11 @@ export class OmniSharpServer {
         }
     }
 
-    private _makeRequest(request: Request) {
+    private _makeRequest(request: Request): number {
+        if (this._state.status !== ServerState.Started) {
+            throw new Error("Tried to make a request when the OmniSharp server wasn't running");
```

Comment:
Should this post to the EventStream instead?

Reply:
I could see it emitting an event. `OmnisharpFailure` isn't what we want because it focuses the OmniSharp Log in the Output pane. It likely needs a new Event type which will be logged by the `OmnisharpDebugModeLoggerObserver`. This could certainly come as a follow up.

Reply:
Okay, I looked into what it would take to make this emit an event instead. The event emitting itself is easy; the problem is that `_makeRequest` needs to return an ID that's used by RequestQueue: https://github.com/OmniSharp/omnisharp-vscode/blob/f1f38b774243526a798a16922d3b8c2f2008a84b/src/omnisharp/requestQueue.ts#L95-L99

So, two options:
1. Keep the `throw` to stop execution.
2. Change `_makeRequest` to return `number | undefined`, emit the event, and don't add the request to the queue if it's undefined.